src/test/codegen/issue-73396-bounds-check-after-position.rs

```diff
@@ -0,0 +1,42 @@
+// no-system-llvm
```

Comment:
```suggestion
// min-llvm-version: 11.0.0
```

Reply:
We always use `no-system-llvm` instead of `min-llvm-version` in tests that check for an optimization, compare https://github.com/rust-lang/rust/search?q=no-system-llvm and https://github.com/rust-lang/rust/search?q=min-llvm-version. Probably because a system LLVM 11 won't be exactly the same as https://github.com/rust-lang/llvm-project.

Reply:
I've never known about that. cc @Mark-Simulacrum could you confirm ?

Reply:
I have no context here, so can't be certain. I suspect we should be using min-llvm-version if the test does not actually need *our* fork of LLVM, which should be rare.

cc @nikic @cuviper

Reply:
I believe the only optimization-relevant change in the fork is related to optimizing away allocations. So apart from that, using min-llvm-version should be fine.

On the other hand we don't actually have any CI coverage for external LLVM versions apart from the minimum one, so maybe using it is unnecessarily risky, as any errors will only be found by distros.

Reply:
When there's nothing specific to rust-llvm backports, I think it's *useful* to let it test on distro LLVM too. I do want to know if my LLVM is broken in some way. I can mask the test downstream if I decide it's something to be ignored.

The challenge is in choosing that `min-llvm-version`. It would obviously be 11.0.0 now, but if the test evolves for something that changes in 12, for instance, we have to remember to update that minimum.